generator/src/main/java/com/scottlogic/deg/generator/generation/string/factorys/RandomStringFactory.java

```diff
@@ -21,57 +21,43 @@
 import dk.brics.automaton.State;
 import dk.brics.automaton.Transition;
 
+import java.util.LinkedList;
 import java.util.List;
 
 public class RandomStringFactory {
 
-    public String createRandomString(
-            String currentString,
-            State state,
-            int minLength,
-            int maxLength,
-            RandomNumberGenerator random) {
+    public String createRandomString(State state, RandomNumberGenerator random) {
+        String currentString = "";
+        List<String> resultOptions = new LinkedList<>();
 
-        if (finishCreating(currentString, state, minLength, maxLength, random)) {
-            return currentString;
+        if (state.isAccept()) {
+            resultOptions.add(currentString);
         }
 
-        List<Transition> transitions = state.getSortedTransitions(false);
+        while (!finishCreating(state)) {
+            List<Transition> transitions = state.getSortedTransitions(false);
 
-        Transition randomTransition = transitions.get(random.nextInt(transitions.size()));
+            Transition randomTransition = transitions.get(random.nextInt(transitions.size()));
 
-        char randomChar = getRandomChar(random, randomTransition);
+            char randomChar = getRandomChar(random, randomTransition);
 
-        return createRandomString(
-            currentString + randomChar,
-            randomTransition.getDest(),
-            minLength,
-            maxLength,
-            random);
-    }
+            currentString = currentString + randomChar;
```

Comment:
Could be simplified with `currentString += getRandomChar(random, randomTransition);`

Reply:
@cdowding-sl I've resolved the issue, but also changed the area to avoid pushing many strings to the heap. Please check over this area before approving.